ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -364,7 +364,7 @@ private List<FollowerInfo> getFollowerInfos(PeerConfiguration peers) {
 
     final Collection<RaftPeer> listeners = conf.getAllPeers(RaftPeerRole.LISTENER);
     if (!listeners.isEmpty()) {
-      addSenders(listeners, placeHolderIndex, false);
+      addSenders(listeners, placeHolderIndex, true);
```

Comment:
@qian0817 , thanks for catching the bug!   Setting `attendVote` to true for listeners does not sound correct since they did not attend the vote.  How about filter them out in `checkStaging()`?

```java
+++ b/ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java
@@ -762,11 +762,15 @@ class LeaderStateImpl implements LeaderState {
       // it is possible that the bootstrapping is done. Then, fallback to UPDATE_COMMIT
       updateCommitEvent.execute();
     } else {
-      final long commitIndex = server.getState().getLog().getLastCommittedIndex();
+      final ServerState state = server.getState();
+      final long commitIndex = state.getLog().getLastCommittedIndex();
+      final Collection<RaftPeerId> listeners = state.getRaftConf().getAllPeers(RaftPeerRole.LISTENER).stream()
+          .map(RaftPeer::getId).collect(Collectors.toSet());
       // check progress for the new followers
       final EnumSet<BootStrapProgress> reports = getLogAppenders()
           .map(LogAppender::getFollower)
           .filter(follower -> !isAttendingVote(follower))
+          .filter(follower -> !listeners.contains(follower.getId()))
           .map(follower -> checkProgress(follower, commitIndex))
           .collect(Collectors.toCollection(() -> EnumSet.noneOf(BootStrapProgress.class)));
       if (reports.contains(BootStrapProgress.NOPROGRESS)) {
```

Reply:
Do you know why it was named attendVote before? This value looks irrelevant to voting and should indicate whether a newly joined node is in the middle of a membership change.

Reply:
You are right -- we may rename it to something like `isCatchingUp`.

Reply:
catchUp is a better name, please review again, thanks.

Reply:
It should be `caughtUp`.  Could you rename it?